re/schedule.go

```diff
@@ -58,50 +106,15 @@ func (s *Schedule) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-// Type can be daily, weekly or monthly.
-type Recurring uint
-
-const (
-	None Recurring = iota
-	Daily
-	Weekly
-	Monthly
-)
-
-func (rt Recurring) String() string {
-	switch rt {
+func (s Schedule) NextDue() time.Time {
+	switch s.Recurring {
 	case Daily:
-		return "daily"
+		return s.Time.AddDate(0, 0, int(s.RecurringPeriod))
 	case Weekly:
-		return "weekly"
+		return s.Time.AddDate(0, 0, int(s.RecurringPeriod)*7)
 	case Monthly:
-		return "monthly"
-	default:
-		return "none"
-	}
-}
-
-func (rt Recurring) MarshalJSON() ([]byte, error) {
-	return json.Marshal(rt.String())
-}
-
-func (rt *Recurring) UnmarshalJSON(data []byte) error {
-	var s string
-	if err := json.Unmarshal(data, &s); err != nil {
-		return err
-	}
-
-	switch s {
-	case "daily":
-		*rt = Daily
-	case "weekly":
-		*rt = Weekly
-	case "monthly":
-		*rt = Monthly
-	case "none":
-		*rt = None
+		return s.Time.AddDate(0, int(s.RecurringPeriod), 0)
 	default:
-		return ErrInvalidRecurringType
+		return time.Time{}
```

Comment:
@dborovcanin Please return the s.Recurring.Time for default 
So if recurring `none` which will run once only